src/Interpreters/executeQuery.cpp

```diff
@@ -563,6 +563,46 @@ void logQueryFinish(
         query_span->addAttributeIfNotZero("clickhouse.written_rows", elem.written_rows);
         query_span->addAttributeIfNotZero("clickhouse.written_bytes", elem.written_bytes);
         query_span->addAttributeIfNotZero("clickhouse.memory_usage", elem.memory_usage);
+
+        if (context)
+        {
+            std::string user_name = context->getUserName();
+            query_span->addAttribute("clickhouse.user", user_name);
+        }
+
+        if (settings[Setting::log_query_settings])
+        {
+            auto changed_settings_names = settings.getChangedNames();
+            for (const auto & name : changed_settings_names)
+            {
+                Field value = settings.get(name);
+                String value_str;
+
+                switch (value.getType())
+                {
+                    case Field::Types::Which::String:
+                        value_str = value.safeGet<String>();
+                        break;
+                    case Field::Types::Which::Int64:
+                        value_str = std::to_string(value.safeGet<Int64>());
+                        break;
+                    case Field::Types::Which::Float64:
+                        value_str = std::to_string(value.safeGet<Float64>());
+                        break;
+                    case Field::Types::Which::Bool:
+                        value_str = value.safeGet<bool>() ? "true" : "false";
+                        break;
+                    case Field::Types::Which::UInt64:
+                        value_str = std::to_string(value.safeGet<UInt64>());
+                        break;
+                    default:
+                        value_str = "Unsupported type: " + std::to_string(static_cast<int>(value.getType()));
+                        break;
+                }
+
+                query_span->addAttribute(name, value_str);
```

Comment:
I would like to suggest adding a prefix(like `clickhouse.setting.`) to the name so that all these query setting names in the span log can be easily distinguished from other attributes.

Reply:
Addressed review comments by adding a "clickhouse.setting." prefix to query setting names